src/AspNet.Security.OAuth.Introspection/Events/BaseIntrospectionContext.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
+
+namespace AspNet.Security.OAuth.Introspection
+{
+    public abstract class BaseIntrospectionContext : BaseControlContext
```

Comment:
I think we shouldn't use `BaseControlContext` for the events called from `HandleAuthenticateAsync` because we can't really control the request processing. `HandleResponse`/`SkipToNextMiddleware` don't really seem super appropriate for token extraction/validation IMHO.